port/common/j9nls.c

```diff
@@ -234,34 +232,30 @@ writeSyslog(struct OMRPortLibrary *portLibrary, uintptr_t flags, const char *for
 		return;
 	}
 
-	/* Attempt to write string to on-stack buffer */
-	COPY_VA_LIST(copyOfArgs, args);
-	numberWritten = portLibrary->str_vprintf(portLibrary, outputBuffer, sizeof(outputBuffer), format, copyOfArgs);
+	/* What is size of buffer required ? str_vprintf(..,NULL,..) result includes the null terminator */
+	uintptr_t bufferSize = portLibrary->str_vprintf(portLibrary, NULL, 0, format, args);
 
-	/* str_vprintf returns number characters written excluding the null terminator */
-	if (sizeof(outputBuffer) > (numberWritten + 1)) {
-		portLibrary->syslog_write(portLibrary, effectiveSyslogFlag, outputBuffer);
-		return;
+	/* use local buffer if possible, allocate a buffer from system memory if local buffer not large enough */
+	if (sizeof(localBuffer) >= bufferSize) {
+		writeBuffer = localBuffer;
+	} else {
+		writeBuffer = portLibrary->mem_allocate_memory(portLibrary, bufferSize, OMR_GET_CALLSITE(), OMRMEM_CATEGORY_PORT_LIBRARY);
 	}
 
-	/* Didn't fit into outputBuffer - find out how long the buffer needs to be */
-	COPY_VA_LIST(copyOfArgs, args);
```

Comment:
All changes requested pushed in latest commit (also, copyright updates to touched files).